### DISCUSSION TITLE
`expand` is missing when all of it's properties are `undefined`

### DISCUSSION THREAD

**Discussion by babakfp** [pocketbase/pocketbase#4346] [Q&A] [answered]
Hi

Am I wrong to expect `expand` property to exist even if `videos` was `undefined`? If `tier` doesn't have any `videos`, then the whole `expand` property is missing. I expect `expand: {}` to be there since I have specified it in the code below `{ expand: "videos" }`.

```ts
const tier: TiersResponse & {
    expand?: {
        videos: VideosResponse[]
    }
} = await locals.pb
    .collection("tiers")
    .getOne(params.id, { expand: "videos" })
```

Maybe a bit related to https://github.com/pocketbase/pocketbase/discussions/4344?

**Comment by ganigeorgiev** [comment, marked as answer]
`expand` is a dynamic field and it is filled only when the user has something to expand (in your case you either don't have permissions or the record doesn't have any videos attached to it).

With the planned refactoring I will consider eventually returning empty `expand` object if there is an `expand` query parameter.
